enterprise/frontend/src/embedding-sdk/components/public/dashboard/SdkDashboard.tsx

```diff
@@ -232,7 +241,29 @@ const SdkDashboardInner = ({
           : onNavigateToNewCardFromDashboard
       }
       onNewQuestion={() => {
-        setRenderMode("queryBuilder");
+        if (isDashboardDirty) {
+          show({
+            title: t`Save your changes?`,
+            message: t`You’ll need to save your changes before leaving to create a new question.`,
+            confirmButtonText: t`Save changes`,
+            onConfirm: async () => {
+              /**
+               * Dispatch the same actions as in the DashboardLeaveConfirmationModal.
+               * @see {@link https://github.com/metabase/metabase/blob/4453fa8363eb37062a159f398050d050d91397a9/frontend/src/metabase/dashboard/components/DashboardLeaveConfirmationModal/DashboardLeaveConfirmationModal.tsx#L30-L34}
+               */
+              setRenderMode("queryBuilder");
+              dispatch(dismissAllUndo());
+              await dispatch(updateDashboardAndCards());
+              // After saving the dashboard, it will exit the editing mode.
+              dispatch(setEditingDashboard(dashboard));
+            },
+            confirmButtonProps: {
```

Comment:
This is a new prop added to the `useConfirmation` hook, because normally it renders `danger` (red) confirmation button.